src/main/java/uk/gov/hmcts/ccd/domain/service/search/global/GlobalSearchResponseTransformer.java

```diff
@@ -74,11 +75,19 @@ public GlobalSearchResponseTransformer(@Qualifier(CachedCaseDefinitionRepository
                                                               final LocationLookup locationLookup) {
         final String jurisdictionId = caseDetails.getJurisdiction();
         final String caseTypeId = caseDetails.getCaseTypeId();
-        final Optional<JurisdictionDefinition> optionalJurisdiction =
-            Optional.ofNullable(caseDefinitionRepository.getJurisdiction(jurisdictionId));
-        final String caseTypeName = Optional.ofNullable(caseDefinitionRepository.getCaseType(caseTypeId))
+
+        final Optional<CaseTypeDefinition> optionalCaseType =
+            Optional.ofNullable(caseDefinitionRepository.getCaseType(caseTypeId));
+        Optional<JurisdictionDefinition> optionalJurisdiction =
+            optionalCaseType.map(CaseTypeDefinition::getJurisdictionDefinition);
+        // clear jurisdiction loaded from case type if ID doesn't match value from case details
+        if (!StringUtils.equals(optionalJurisdiction.map(JurisdictionDefinition::getId).orElse(null), jurisdictionId)) {
+            optionalJurisdiction = Optional.empty();
+        }
```

Comment:
Do we actually need this? Is there any scenario it can happen, and if so is this the correct thing to do anyway?

Reply:
I agree it is unlikley to happen but aded the scenario and unit tests as a belt and braces, I will leave in as it passes tests.